### PR TITLE
Homework Mock

### PR DESCRIPTION
**Pull request by kari0036**

### CODE REVIEW

File: test/PizzeriaTest.cpp [hunk 1]
```diff
@@ -63,5 +83,45 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
...
+    // When
+    auto orderId = domino.makeOrder(pizzas);
+    auto price = domino.calculatePrice(orderId);
+    domino.bakePizzas(orderId);
+    domino.completeOrder(orderId);
+
+    // Then
+    ASSERT_EQ(price, mock1Price + mock2Price);
+
+    delete ps;
```
Comment (MaciejSalamonski) [inline test/PizzeriaTest.cpp:126]: +1

**Review by MaciejSalamonski — APPROVED** [review]
Very nice 👍

File: test/PizzeriaTest.cpp [hunk 2]
```diff
@@ -1,60 +1,80 @@
...
+    delete pd;
 }
 
 TEST_F(PizzeriaTest, completeOrderWithStubPizza)
 {
     // Given
-    Pizzas pizzas = {new PizzaStub{"STUB"}};
+    Pizza* ps = new PizzaStub{"stub"};
+    Pizzas pizzas = {ps};
+    EXPECT_CALL(dt, sleep_for).Times(1);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:63]:
```suggestion
    EXPECT_CALL(dt, sleep_for(ps.getBakingTime())).Times(1);
```

File: test/PizzeriaTest.cpp [hunk 3]
```diff
@@ -63,5 +83,45 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
...
+
+    EXPECT_CALL(pm1, getPrice()).WillOnce(Return(mock1Price));
+    EXPECT_CALL(pm1, getName()).WillOnce(Return(mock1Name));
+    EXPECT_CALL(pm1, getBakingTime()).WillOnce(Return(mock1BakingTime));
+
+    EXPECT_CALL(pm2, getPrice()).WillOnce(Return(mock2Price));
+    EXPECT_CALL(pm2, getName()).WillOnce(Return(mock2Name));
+    EXPECT_CALL(pm2, getBakingTime()).WillOnce(Return(mock2BakingTime));
+
+    EXPECT_CALL(tm, sleep_for).Times(3);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:115]:
```suggestion
    EXPECT_CALL(tm, sleep_for(ps.getBakingTime())).Times(1);
    EXPECT_CALL(tm, sleep_for(mock1BakingTime)).Times(1);
    EXPECT_CALL(tm, sleep_for(mock2BakingTime)).Times(1);
```

**Review by ziobron — CHANGES_REQUESTED** [review]
Almost good. Only one change needed.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and have nothing to add.

File: test/PizzeriaTest.cpp [hunk 4]
```diff
@@ -60,7 +60,7 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
     // Given
     Pizza* ps = new PizzaStub{"stub"};
     Pizzas pizzas = {ps};
-    EXPECT_CALL(dt, sleep_for).Times(1);
+    EXPECT_CALL(dt, sleep_for(ps.getBakingTime())).Times(1);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:63]:
```suggestion
    EXPECT_CALL(dt, sleep_for(ps->getBakingTime())).Times(1);
```
ps is a pointer. Sorry for my misleading tip, but CI detected that this code does not compile

File: test/PizzeriaTest.cpp [hunk 5]
```diff
@@ -112,7 +112,9 @@ TEST(PizzeriaTestMain, shouldCalculateOrderLikeInMain)
     EXPECT_CALL(pm2, getName()).WillOnce(Return(mock2Name));
     EXPECT_CALL(pm2, getBakingTime()).WillOnce(Return(mock2BakingTime));
 
-    EXPECT_CALL(tm, sleep_for).Times(3);
+    EXPECT_CALL(tm, sleep_for(ps.getBakingTime())).Times(1);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:115]:
```suggestion
    EXPECT_CALL(tm, sleep_for(ps->getBakingTime())).Times(1);
```

**Review by ziobron — APPROVED** [review]
Small changes needed (the code does not compile), but I accept it :)

**Comment by ziobron** [comment]
{ "requirement_ids" : [1119, 1119, 1119, 3524, 3525, 3526] }
3 * 3 XP za punktualność (1119)
10 XP za TimeMock w Pizzas (3524)
15 XP za nowy przypadek testowy w Pizzas (3525)
5 XP za workflow w Pizzas (3526)

**Pull request closed without merging**